logins.js

```diff
@@ -9,107 +9,111 @@ var oauth_service = 'audience:server:client_id:848232511240-7so421jotr2609rmqakc
 var app = 'com.nianticlabs.pokemongo';
 var client_sig = '321187995bc7cdc2b5fc91b11a96e2baa8602c62';
 
-module.exports = {
-    PokemonClub: function (user, pass, self, callback) {
-        var options = {
+
+var PokemonClub = function (user, pass, self, callback) {
+    var options = {
+        url: login_url,
+        headers: {
+            'User-Agent': 'niantic'
+        }
+    };
+
+    self.request.get(options, function (err, response, body) {
+        if (err) {
+            return callback(err, null);
+        }
+
+        var data = null;
+        try {
+          data = JSON.parse(body);
+        }catch(err){
+          return callback(err, null);
+        }
+
+        options = {
             url: login_url,
+            form: {
+                'lt': data.lt,
+                'execution': data.execution,
+                '_eventId': 'submit',
+                'username': user,
+                'password': pass
+            },
             headers: {
                 'User-Agent': 'niantic'
             }
         };
 
-        self.request.get(options, function (err, response, body) {
-            var data;
-
-            if (err) {
+        self.request.post(options, function (err, response, body) {
+            //Parse body if any exists, callback with errors if any.
+            if(err) {
                 return callback(err, null);
             }
-            
-            try {
-              data = JSON.parse(body);
-            }catch(err){
-              return callback(err, null);
+
+            if (body) {
+                var parsedBody = JSON.parse(body);
```

Comment:
So I just refactore the code. Didn't actually change anything . Github's diff is off because i pulled out the two objects (`GoogleAccount` and `PokemonClub`)